src/tools/wasm-split.cpp

```diff
@@ -245,8 +261,195 @@ void parseInput(Module& wasm, const WasmSplitOptions& options) {
   options.applyFeatures(wasm);
 }
 
+// Add a global monotonic counter and a timestamp global for each function, code
+// at the beginning of each function to set its timestamp, and a new exported
+// function for dumping the profile data.
+struct FuncInstrumentor : public Pass {
```

Comment:
This name seems too generic - how about `TimeInstrumentor`?

Reply:
`Func` is supposed to differentiate it from a hypothetical future `BlockInstrumenter`. But given that this is the only instrumentation in the tool right now, how about doubling down on the generality and changing the name to "Instrumenter"? This is entirely unambiguous in the context of wasm-split.

Reply:
sgtm. I didn't realize the intended comparison to blocks.